File: python/day_1.py
# Part 1
with open('../input/input_1.txt', 'r') as f:
    elves = {}
    calories = 0
    elf = 1
    for line in f:
        if line != '\n':
            calories += int(line)
        else:
            elves[elf] = calories
            calories = 0
            elf += 1

elves_ranked = dict(sorted(elves.items(), key=lambda item: item[1], reverse=True))
print(list(elves_ranked.values())[0])

# Part 2
print(sum(list(elves_ranked.values())[0:3]))

File: scala/day_1.scala
import scala.io.Source

object day_1 extends App {

    val file: String = "../input/input_1.txt"
    var elves = Set()
    var calories: Int = 0
    var elf: Int = 1

    for (line <- Source.fromFile(file).getLines.mkString) {

        if (line != "\n\n") {
            println(line)
        }

    }

}